### PR TITLE
Shapes

### PR DESCRIPTION
**Pull request by evgeniyamelone**
Write a demo program in which create an array of Shapes and add at least one Circle, Rectangle and Square object in it. Iterate through the array elements and for each object:
- print object details using toString method;
- print out perimeter or area based on the object class.

### CODE REVIEW

File: src/main/java/Sailboat.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+public class Sailboat extends Boat {
+    public void move() {
```
Comment (justYP) [inline src/main/java/Sailboat.java:2]: `@Override` annotation is missing

File: src/main/java/Boat.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public void setLenght(int len) {
+        lenght = len;
+    }
+
+    public int getLength() {
+        return lenght;
+    }
+
+    public void move() {
+        System.out.print("Дрейф ");
```
Comment (justYP) [inline src/main/java/Boat.java:13]: Extra space after `Дрейф`

File: src/main/java/circle/Circle.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package circle;
+
+public class Circle {
+    private double radius;
+    private String colour;
+    final double PI = 3.14;
+
+    //constructors
```
Comment (justYP) [inline src/main/java/circle/Circle.java:7]: Minor: This kind of comments is useless

File: src/main/java/circle/Circle.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    String getColour() {
+        return colour;
+    }
+
+    //setters
+    public void setRadius(double radius) {
+        this.radius = radius;
+    }
+
+    public void setRadius(String colour) {
```
Comment (justYP) [inline src/main/java/circle/Circle.java]: Invalid name of the method

File: src/main/java/circle/Circle.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        this.colour = colour;
+    }
+
+    //area of​a circle
```
Comment (justYP) [inline src/main/java/circle/Circle.java:39]: A typo in the comment. 
P.S. this comment is useless as well

File: src/main/java/circle/Circle.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    public double getArea() {
+        double radius = getRadius();
+        return PI * radius * radius;
+    }
+
+    public String toString() {
+        return "Circle parameters: \n\tradius: " + getRadius() + "\n\tcolour: " + getColour() + "\n\tarea: " + getArea();
```
Comment (justYP) [inline src/main/java/circle/Circle.java]: Minor: Please have a look at more convenient `toString `implementation
https://www.baeldung.com/java-tostring

File: src/main/java/circle/Circle.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        this.colour = colour;
+    }
+
+    //area of​a circle
+    public double getArea() {
+        double radius = getRadius();
+        return PI * radius * radius;
+    }
+
+    public String toString() {
```
Comment (justYP) [inline src/main/java/circle/Circle.java:46]: `@Override` annotation is missing

File: src/main/java/circle/Circle.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
+package circle;
+
+public class Circle {
+    private double radius;
+    private String colour;
+    final double PI = 3.14;
```
Comment (justYP) [inline src/main/java/circle/Circle.java]: You can use `Math.PI` instead

File: src/main/java/circle/Cylinder.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
+package circle;
+
+public class Cylinder extends Circle {
+    private double height;
+    final double PI = 3.14;
```
Comment (justYP) [inline src/main/java/circle/Cylinder.java]: This constant is already defined in the parent class as package-local so redefining is redundant. And you can use Math.PI instead

File: src/main/java/shapes/Shape.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    void setFilled(boolean filled) {
+        this.filled = filled;
+    }
+
+
+    @Override
+    public String toString() {
+        String isNot = "filled";
```
Comment (justYP) [inline src/main/java/shapes/Shape.java]: Not really good name for the variable. `isNot `can be "filled" or can be "not filled" - it may confuse.

Comment (justYP) [reply src/main/java/shapes/Shape.java]: Another way to do it:
`String filledText = isFilled() ? "filled" : "not filled";`

**Review by justYP — CHANGES_REQUESTED** [review]

File: src/main/java/circle/Circle.java (hunk 8 above)
Comment (evgeniyamelone) [reply src/main/java/circle/Circle.java]: Fixed.

File: src/main/java/circle/Circle.java (hunk 7 above)
Comment (evgeniyamelone) [reply src/main/java/circle/Circle.java]: Fixed.

File: src/main/java/circle/Cylinder.java (hunk 9 above)
Comment (evgeniyamelone) [reply src/main/java/circle/Cylinder.java]: Fixed.

File: src/main/java/circle/Circle.java (hunk 4 above)
Comment (evgeniyamelone) [reply src/main/java/circle/Circle.java]: > Invalid name of the method

Why? Not clear

File: src/main/java/circle/Circle.java (hunk 7 above)
Comment (evgeniyamelone) [reply src/main/java/circle/Circle.java:46]: > `@Override` annotation is missing

fixed.

File: src/main/java/shapes/Shape.java (hunk 10 above)
Comment (evgeniyamelone) [reply src/main/java/shapes/Shape.java]: > Another way to do it:
> `String filledText = isFilled() ? "filled" : "not filled";`

fixed.

**Pull request merged**